### PR TITLE
i172: sign in with email integration test

### LINKED ISSUE #172: Add integration tests for Auth user flows
Initial Plan: 
- Use the live AuthService (no need to mock it, Nick might look at alternative solutions down the track) 
- Point the Storage to a special bucket for tests (but probably not necessary for Auth tests actually as the Storage isn't used anywhere in the Auth section) 
- Connect to the other firebase emulators (Firestore, etc.) for the integration tests 
- Write tests for each user flow that makes sense 
- Check in if any doubts 

Tasks 
- [ ] Check that signing in with google sets the profile pic url
  - difficult as there is no local emulator for auth service yet 
- [ ] test that a new sign in correctly adds user details to database 
  - I think this is not going to work?  as there is no firebase auth emulator - need to discuss 
- [ ] make sure to test things that need to be tested on a real device - interacting with plugins is the main one but there are probably others 

- Note: this is meant as a starting point, as a place to put ideas for what to test as they come up. Might be better in the issue 

#### Definitely do 
- [ ] test correct sign in button shows on each platform 
#### Maybe do 
- [ ] test using Sign In with Apple on device that has no Apple Id setup

### PR DESCRIPTION
**Pull request by tamari-gray**
This was made as a starting point for #172 . Does not fully fix the issue

### CODE REVIEW

File: test_driver/auth.dart [hunk 1]
```diff
@@ -0,0 +1,34 @@
+import 'dart:async';
```
Comment (tamari-gray) [inline test_driver/auth.dart:1]: created flutter driver without redux devtools

File: test_driver/auth_test.dart [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import 'package:flutter_driver/flutter_driver.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('Auth section', () {
+    // create finders
+    final emailAuthButton = find.byType('EmailOptionsFAB');
```
Comment (tamari-gray) [inline test_driver/auth_test.dart:7]: felt kind of messy to create so many finders up the top, open to suggestions for cleaning her up or leaving if shes alg. I can see the finders list growing alot with a bigger integration test, but maybe worry about that potential clutter when it comes?

File: test_driver/mocks/auth_service_mocks.dart [hunk 3]
```diff
@@ -2,8 +2,12 @@ import 'dart:async';
```
Comment (tamari-gray) [inline test_driver/mocks/auth_service_mocks.dart:2]: made sign in streams update the onAuthStateChanged stream with a signed in user

File: test_driver/mocks/utils/verify_dispatch_middleware.dart [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+/// [Middleware] that keeps track of actions that have been dispatched, for
+/// testing.
+///
+/// [actions] is a static/class member so that it can be accessed in the
+/// anonymous funciton that is created in the class initializer
+class VerifyDispatchMiddleware extends TypedMiddleware<AppState, ReduxAction> {
+  static List<ReduxAction> actions = [];
+  VerifyDispatchMiddleware()
+      : super((store, action, next) async {
+          next(action);
```
Comment (tamari-gray) [inline test_driver/mocks/utils/verify_dispatch_middleware.dart:14]: do we want the next method taken out of here too?
i just noticed it

**Review by tamari-gray — COMMENTED** [review]
LTRB (LGTM)

File: test_driver/auth.dart (hunk 1 above)
Comment (nickmeinhold) [reply test_driver/auth.dart:1]: nice!

File: test_driver/auth_test.dart (hunk 2 above)
Comment (nickmeinhold) [reply test_driver/auth_test.dart:7]: Yeah I feel you - the only thing off the top of my head I can think of would be to wrap them in a class or just as top level variables in a separate file.  I agree though, worry about it when it cones :-)

File: test_driver/firebase_emulators.dart [hunk 5]
```diff
@@ -7,7 +7,7 @@ import 'package:flutter/material.dart';
 import 'package:flutter_driver/driver_extension.dart';
 import 'package:redux_remote_devtools/redux_remote_devtools.dart';
 
-Future<void> main() async {
+void main() {
```
Comment (nickmeinhold) [inline test_driver/firebase_emulators.dart:10]: Why so async?

File: test_driver/mocks/auth_service_mocks.dart (hunk 3 above)
Comment (nickmeinhold) [reply test_driver/mocks/auth_service_mocks.dart:2]: The FakeAuthService is a boss.

File: test_driver/mocks/utils/verify_dispatch_middleware.dart (hunk 4 above)
Comment (nickmeinhold) [reply test_driver/mocks/utils/verify_dispatch_middleware.dart:14]: let's just delete the whole thing!  I might make a clean up PR after this goes in, just to clean up the mock files I added messily, so I can remove it then :-)

**Review by nickmeinhold — APPROVED** [review]
Beautiful!  A sweet set of integration tests!

**Pull request merged**